### DISCUSSION TITLE
Change default Pull Request filter

### DISCUSSION THREAD

**Discussion by aaelberg** [community/community#51751] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I'd like to be able to change the default filter of Pull Requests page.
Default is "is:pr is:open " and although I can change the filter, each time I refresh the page it gets back to the default.

Thanks

**Comment by tom-id-au** [comment]
Very keen on this

**Comment by DaanKorver** [comment]
Would be really nice, especially when working in a mono repo with lots of Pull Requests and issues

**Comment by osmestad** [comment]
Hiding drafts is for instance something I'd like to switch on: `is:pr is:open draft:false` (surprised it is not default)

**Comment by ilyakonrad** [comment]
I think it's an essential quality-of-life feature. You should be able to specify a default search query in "Pull requests".
My use-case is to sort PRs by recently updated. Some PRs are open for 1 day, whereas some other, bigger PRs may hang there for months.
In the meantime dependabot-like tools create tons of small PRs that push the important ones down the list, even though they are actively being worked on.

**Comment by ErikStocktonTR** [comment]
Yes, please.  Really need the ability to set default filters.

**Comment by svanscho** [comment]
Would love this!

**Comment by rwbeauchemin** [comment]
Yes, would love this!

**Comment by rwbeauchemin** [comment]
https://github.com/rbeauchemin/github-filter
I made a little extension this morning to at least make this a 1-click filter for Chrome users. Feel free to take and re-use! Don't forget to check the source code before adding this extension to stay safe (it's very little to review)

**Comment by luisrayas3** [comment]
Default would be great, and having custom presets in the drop-down would also be nice. I currently work around this with browser bookmarks.

**Comment by hosua** [comment]
Would be nice, as being able to hide draft PRs by default would be handy.

**Comment by blakethomp** [comment]
I was looking for this functionality and came across this thread, I decided to create a very simple tampermonkey script to apply my preferred filters to `/pulls` links

```
// ==UserScript==
// @name         Default PR Filters
// @namespace    http://tampermonkey.net/
// @version      2024-08-13
// @description  Append preferred PR filter query param to default /pulls hrefs.
// @author       You
// @match        https://github.com/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=github.com
// @grant        none
// ==/UserScript==

(function() {
    'use strict';
    const prLinks = document.querySelectorAll('[href$="/pulls"]');
    prLinks.forEach((linkEl) => {
        linkEl.setAttribute('href', `${linkEl.getAttribute('href')}?q=is%3Apr+is%3Aopen+draft%3Afalse`);
    });
})();
```

**Comment by shirley-papuga** [comment]
100% support this change in functionality!

**Comment by twenty12** [comment]
As someone who reviews a lot of PRs and has very particular label filtering setup, this would be great!

**Comment by ridafkih** [comment]
I liked @blakethomp's band-aid fix, but didn't work for me for some URLs and depending on the source of the anchor tag, how the `href` is created and updated, or those with pre-defined query parameters. I made it so that it also appends `draft:false` to the `q` parameter on anchors with the `href` attribute already defined.

I use mine as an Arc Boost, but there's no reason it wouldn't also work as a ViolentMonkey/TamperMoney script. You can adjust the new query to your liking.

```js
const DEFAULT_FILTERS = ["is:pr", "is:open", "draft:false"];

const updateAnchor = (element) => {
    const href = element.getAttribute("href")
    if (!href || !href.startsWith("/")) return
    const url = new URL(href, window.location.origin)
    if (!url.pathname.endsWith("/pulls")) return
  
    const query = url.searchParams.get("q")?.split(" ")

    if (!query) {
        url.searchParams.set("q", DEFAULT_FILTERS.join(" "))
        element.setAttribute("href", url.href)
        return
    }

    if (!query.includes("is:open") || query.includes("draft:false")) return
    query.push("draft:false")
    url.searchParams.set("q", query.join(" "))
    element.setAttribute('href', url.href)
}

const updateAll = () => {
    const elements = document.querySelectorAll('a[href]')
    for (const element of elements) updateAnchor(element)
}

const observe = (mutations) => {
    for (const mutation of mutations) {
        switch (mutation.type) {
            case "childList":
                for (const node of mutation.addedNodes) {
                    if (node.nodeType !== Node.ELEMENT_NODE) continue
                    if (node.tagName === 'A') updateAnchor(node)
                    for (const anchor of node.querySelectorAll('a[href]')) updateAnchor(anchor)
                }
                break;
            case "attributes":
                if (mutation.attributeName !== "href") break
                updateAnchor(mutation.target)
                break
            default:
        }
    }
}

const observer = new MutationObserver(observe)

observer.observe(document.documentElement, { 
    childList: true, 
    subtree: true,
    attributes: true,
    attributeFilter: ['href']
})

document.addEventListener("DOMContentLoaded", updateAll)
```

**Reply by vipulwairagade** [reply]
How do I use this exactly ?

**Reply by ridafkih** [reply]
@vipulwairagade You can paste the code into a Tampermonkey script set to run on GitHub, or "Arc Boost"

**Comment by jrencz** [comment]
It would be nice to be able to set default PRs filter to a `is:pr is:open -is:draft`

**Comment by vieteh** [comment]
for us is the ability to only filter pull requests to the default branch

**Comment by angelinashepherd** [comment]
PLEASE this would be legendary

**Comment by daxaxelrod** [comment]
I’d really appreciate this. Draft PRs are often used just to inspect the diff, and reviews usually aren’t needed until they’re marked as ready

**Comment by zachfey** [comment]
The new UI updates are great. Disappointed that being able to save a default filter was not included! Please at least hide drafts by default..

**Comment by Tristan-zimmerman** [comment]
Adding another voice to asking for this feature. I'd particularly like to be able to have a default sort so that the PRs appear with oldest first instead of last as older PRs tend to get buried by the new: `is:open is:pr sort:created-asc `. Having a way to set a default filter string for a repo or org would be tops.

**Comment by LeonardoCortelim** [comment]
Very keen on this

**Comment by irl-segfault** [comment]
pls

**Comment by ryanalvita** [comment]
Would be nice if this is implemented, either draft hidden as default or being able to create our own 'default' filter

**Comment by chrisdl** [comment]
I too would appreciate this functionality 🙇

**Comment by recursivefunk** [comment]
Would also love this. Any movement?

**Comment by zhztheplayer** [comment]
+1, love this.

**Comment by ironlam** [comment]
+1 , it would be a great improvement.

**Comment by joeyJsonar** [comment]
please implement

**Comment by metuan** [comment]
+1, it would be a helpful!

**Comment by markwaddle** [comment]
+1, i would set the default to `is:pr is:open -is:draft` for my entire team

**Comment by okolobaxa** [comment]
Almost 3 years, but still very helpful

**Comment by hoi-leasi** [comment]
+1, please, this would be huge!

**Comment by alliecatowo** [comment]
shocked this isn't a thing yet, +1

**Comment by edgartheunready** [comment]
In this multi-agent, multi-worktree world we live in now, it would be wonderful to support this. That or create some sort of "hidden draft" state.

I want to be able to create WIP PRs, and I want to not contribute to repo noise.

**Comment by sylvainb-unito** [comment]
Big +1 !

**Comment by pruthvip15** [comment]
+1

**Comment by DavidPhilip** [comment]
+1

**Comment by oliverpaulsen** [comment]
+1

**Comment by chrisjenx** [comment]
Github not adding features that are actually usefull since.... will 2023 or longer is most cases!

**Comment by beniclouds** [comment]
+1

**Comment by tylertylerday** [comment]
If anyone is interested, I made a little chrome and firefox extension that solves this. You can create your own quick filters, and set a default.
"Github Quick Filters"
[chrome](https://chromewebstore.google.com/detail/github-pr-quick-filters/mnebjmklpamjgpmogmgdecophcacaema)
[firefox](https://addons.mozilla.org/en-US/firefox/addon/github-pr-quick-filters/)

**Comment by sushie1984** [comment]
+ 1
